### ISSUE TITLE
ticketbuyer errors

### ISSUE THREAD

**Issue by peterzen** [decred/dcrwallet#1617]
I'm running the 2 ticketbuyers setup described on `cspp.decred.org`, the new mixed buyer (buyer2) is printing the below errors.  The `insufficient balance:: txauthor.NewUnsignedTransaction` messages have been in the logs for several days as I presume there weren't sufficient eligible inputs until today.

`read *messages.BR (dicemix.go:233): unexpected EOF` appeared today the first time.

I'm not sure if the errors are related or not, apologies for lumping them together if not.

dcrwallet `release-v1.5`, go1.13.4 linux/amd64

```
2019-11-25 16:39:04.160 [INF] SYNC: Connected block 000000000000000016ab65aaa8d67bc4337b5b4d352aaf385e3c5aebb2575001, height 400325, 0 wallet transaction(s)
2019-11-25 16:39:04.165 [INF] WLLT: Returning address (account=2 branch=1 child=0)
2019-11-25 16:39:04.596 [ERR] TKBY: One or more tickets could not be purchased: wallet.PurchaseTicketsContext: insufficient balance:: txauthor.NewUnsignedTransaction
2019-11-25 16:41:21.332 [DBG] WLLT: Updating processed txs block marker to 00000000000000001a3191910fc21efcabe95b3f084031e98b05766e3b6e056a
2019-11-25 16:41:21.760 [INF] SYNC: Connected block 00000000000000001a3191910fc21efcabe95b3f084031e98b05766e3b6e056a, height 400326, 0 wallet transaction(s)
2019-11-25 16:41:21.766 [INF] WLLT: Returning address (account=2 branch=1 child=0)
2019-11-25 16:41:22.186 [ERR] TKBY: One or more tickets could not be purchased: wallet.PurchaseTicketsContext: insufficient balance:: txauthor.NewUnsignedTransaction
2019-11-25 16:47:02.484 [DBG] WLLT: 10.11.40.6:46132: read *messages.BR (dicemix.go:233): unexpected EOF
2019-11-25 16:47:02.908 [ERR] TKBY: One or more tickets could not be purchased: wallet.PurchaseTicketsContext: 10.11.40.6:46132: read *messages.BR (dicemix.go:233): unexpected EOF
2019-11-25 16:48:04.242 [DBG] WLLT: Updating processed txs block marker to 0000000000000000121f442cfd7e76ad7919e7844b50af540b76e4e7831b4c93
```

```
2019-11-26 03:51:38.349 [INF] WLLT: Dialed CSPPServer X.Y.W.Z:49972 -> 45.63.79.180:5760
2019-11-26 03:53:12.114 [DBG] WLLT: Updating processed txs block marker to 00000000000000000a14a72d9d7a63b899472379a0ed8b195711698b79c8bce2
2019-11-26 03:53:12.542 [INF] SYNC: Connected block 00000000000000000a14a72d9d7a63b899472379a0ed8b195711698b79c8bce2, height 400461, 0 wallet transaction(s)
2019-11-26 03:53:12.548 [INF] WLLT: Returning address (account=2 branch=1 child=0)
2019-11-26 03:53:12.966 [ERR] TKBY: One or more tickets could not be purchased: wallet.PurchaseTicketsContext: insufficient balance:: txauthor.NewUnsignedTransaction
2019-11-26 03:55:10.391 [DBG] WLLT: Updating processed txs block marker to 0000000000000000193f7b732124a699c4b6252276e88126fb46e5f0d78f89e2
2019-11-26 03:55:10.816 [INF] SYNC: Connected block 0000000000000000193f7b732124a699c4b6252276e88126fb46e5f0d78f89e2, height 400462, 0 wallet transaction(s)
2019-11-26 03:55:10.820 [ERR] WLLT: Failed to watch for future address usage after publishing transaction: context canceled
2019-11-26 03:55:10.820 [ERR] WLLT: Failed to watch for future addresses after ticket purchases: context canceled
2019-11-26 03:55:10.820 [ERR] TKBY: One or more tickets could not be purchased: wallet.PurchaseTicketsContext: context canceled
```

**Comment by jrick** [comment]
An EOF on the BR message (the first message sent to the client by the server) may be due to the server restarting or network loss somewhere in the network path.  It used to be that submitting an invalid mix to the server (double spends, etc) would cause the server to immediately terminate the connection, which the client would observe an EOF, but it no longer does that and sends an error code instead.

The insufficient balance errors during ticket purchasing should be silenced (https://github.com/decred/dcrwallet/blob/v1.5.0-rc2/ticketbuyer/tb.go#L151-L158), but essentially what is happening is that because of an ongoing mix request, your available balance to create a new request with is not enough because the outputs are already selected.  I'll take a look into why those errors are still printing.

**Comment by peterzen** [comment]
The `unexpected EOF` didn't reoccur since yesterday.

**Comment by peterzen** [comment]
Don't know if it's relevant - buyer2 hasn't been able to buy any tickets so far.  The configuration:

```
csppserver=cspp.decred.org:5760 
csppserver.ca=/home/dcrwallet/.dcrwallet/cspp.decred.org.pem
enableticketbuyer=true
purchaseaccount=mixed 
mixedaccount=mixed/1 
changeaccount=unmixed 
ticketbuyer.votingaccount=voting 
mixchange=true
```

**Comment by peterzen** [comment]
> The insufficient balance errors during ticket purchasing should be silenced (https://github.com/decred/dcrwallet/blob/v1.5.0-rc2/ticketbuyer/tb.go#L151-L158), but essentially what is happening is that because of an ongoing mix request, your available balance to create a new request with is not enough because the outputs are already selected. I'll take a look into why those errors are still printing.

Is it conceivable that the mix request is ongoing for such a long time and those 45 outputs in `mixed` are all blocked on buyer2?  In total, there's ~480 DCR spendable in the account.  As a test, I sent a small amount from `mixed` to a new address back to `mixed`, which then correctly got sent to CSPP and back (`Completed CoinShuffle++ mix of output xxxxx:1 in transaction aaabbbccc`).

I wonder whether something is wrong in my wallet setup or this is the expected behavior.

Just to recap, the error message is
```
[ERR] TKBY: One or more tickets could not be purchased: wallet.PurchaseTicketsContext: insufficient balance:: txauthor.NewUnsignedTransaction
```

**Comment by peterzen** [comment]
Closing this and opening a new issue with cleaned up logs and updated findings.
